diff --git a/jw_player.module b/jw_player.module
index b2b483d..656f96c 100644
--- a/jw_player.module
+++ b/jw_player.module
@@ -58,7 +58,10 @@ function jw_player_field_formatter_info() {
   $formatters = array(
     'jw_player' => array(
       'label' => t('JW player'),
-      'field types' => array('file'),
+      'field types' => array(
+        'file',
+        'video',
+      ),
       'settings' => array(
         'jwplayer_preset' => '',
       ),
