--- jw_player.module.orig	2011-11-14 21:48:46.000000000 +0800
+++ jw_player.module	2011-11-14 21:48:51.000000000 +0800
@@ -130,7 +130,7 @@
 function jw_player_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
   $element = array();
 
-  if ($display['type'] == 'jw_player') {
+  if ($display['type'] == 'jw_player' && count($items) > 0) {
     // Process files for theme functions
     $files = array();
     foreach ($items as $delta => $item) {
